src/modules/navigator/rtl.cpp

```diff
@@ -43,43 +43,71 @@
 #include "navigator.h"
 #include <dataman/dataman.h>
 
-
 static constexpr float DELAY_SIGMA = 0.01f;
 
+using namespace time_literals;
+
 RTL::RTL(Navigator *navigator) :
 	MissionBlock(navigator),
 	ModuleParams(navigator)
 {
 }
 
-void
-RTL::on_inactivation()
+void RTL::on_inactivation()
 {
 	if (_navigator->get_precland()->is_activated()) {
 		_navigator->get_precland()->on_inactivation();
 	}
 }
 
-void
-RTL::on_inactive()
+void RTL::on_inactive()
 {
 	// Reset RTL state.
 	_rtl_state = RTL_STATE_NONE;
 
 	find_RTL_destination();
-
 }
 
-void
-RTL::find_RTL_destination()
+void RTL::find_RTL_destination()
 {
-	// get home position:
-	home_position_s &home_landing_position = *_navigator->get_home_position();
-	// get global position
-	const vehicle_global_position_s &global_position = *_navigator->get_global_position();
+	// require updated global position
+	vehicle_global_position_s global_position;
+
+	if (!_vehicle_global_position_sub.update(&global_position)) {
+		return;
+	}
+
+	// require valid home position
+	bool home_updated = _home_position_sub.updated();
+	home_position_s home_landing_position{};
+	_home_position_sub.copy(&home_landing_position);
+
+	if (home_landing_position.timestamp == 0 || !home_landing_position.valid_hpos) {
+		return;
+	}
+
+	if (!home_updated) {
+		// don't update RTL destination faster than 1 Hz
+		if (hrt_elapsed_time(&_destination_check_time) < 1_s) {
+			return;
+		}
+
+		// don't update RTL destination if position hasn't changed
+		float d = get_distance_to_next_waypoint(_destination_check_last_lat, _destination_check_last_lon, global_position.lat,
+							global_position.lon);
```

Comment:
I think this is not exactly cheap CPU-wise.

Reply:
It's not, but it doesn't matter at < 1 Hz. We should add geo it to the microbenchmarks.